File: app/src/main/java/com/yalantis/data/source/repository/RepositoryDataSource.java
package com.yalantis.data.source.repository;

import android.support.annotation.NonNull;

import com.yalantis.data.Repository;

import java.util.List;

import rx.Single;

/**
 * Created by irinagalata on 12/1/16.
 */

interface RepositoryDataSource {

    Single<List<Repository>> getRepositories(@NonNull String organization);

    void saveRepositories(List<Repository> repositories);

    void clearRepositories();

    boolean isEmpty();

}

File: app/src/main/java/com/yalantis/data/source/repository/RepositoryDataSource.kt
package com.yalantis.data.source.repository

import com.yalantis.data.Repository
import io.reactivex.Single


/**
 * Created by irinagalata on 12/1/16.
 */

internal interface RepositoryDataSource {

    fun getRepositories(organization: String): Single<List<Repository>>

    fun saveRepositories(repositories: List<Repository>)

    fun clearRepositories()

    fun isEmpty(): Boolean

}
